### PR TITLE
trivial: treewide: random noop cleanups

### PR DESCRIPTION
**Pull request by oxij**
###### Motivation for this change

OCD.

###### Things done

- [X] Nothing changes.

### CODE REVIEW

**Comment by oxij** [comment]
@Ericson2314 So, do you want me to change something here?

**Comment by Ericson2314** [comment]
Actually no the nix parameter cannot be removed, but the environment variable can. Anyways, just drop those TODO removes from this, and I'll clean up GHC separately.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -16022,7 +16022,7 @@ with pkgs;
       ImageCaptureCore GSS ImageIO;
   };
 
-  emacsPackagesGen = emacs: self: let callPackage = newScope self; in rec {
+  emacsPackagesFor = emacs: self: let callPackage = newScope self; in rec {
```
Comment (Ericson2314) [inline pkgs/top-level/all-packages.nix]: Does this need an alias.nix entry? Not sure whether this was intended to be user-exposed.

**Comment by oxij** [comment]
Done.

I have a bunch more cleanups (that need a mass rebuild, so they'll come later) and when doing those I had a problem with your TODO in `openssl` expression. Please look at that one too.

**Comment by oxij** [comment]
> Does this need an alias.nix entry? Not sure whether this was intended to be user-exposed.

Yes it was, you'd have to use it if you want to override `emacs` in your config. Like, say, enable gtk3.

**Comment by Ericson2314** [comment]
OK all makes sense. Looks good.

**Review by Ericson2314 — APPROVED** [review]

**Pull request merged**